Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelperTest.java

```diff
@@ -0,0 +1,257 @@
+		eventA.setDirection("outbound");
+		AdhocQueryRequestMessageType msg = new AdhocQueryRequestMessageType();
+		AdhocQueryRequest adhocQueryRequest = new AdhocQueryRequest();
+		SlotType1 slotType1 = new SlotType1();
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		ValueListType valueListType = new ValueListType();
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+		AdhocQueryType adhocQueryType = new AdhocQueryType();
+		adhocQueryType.getSlot().add(slotType1);
+
+		adhocQueryRequest.setAdhocQuery(adhocQueryType);
+		msg.setAdhocQueryRequest(adhocQueryRequest);
+		AssertionType assertionTypeTest = new AssertionType();
+		assertionTypeTest.setMessageId(MESSAGE_ID);
+		HomeCommunityType homeCom = new HomeCommunityType();
+		homeCom.setHomeCommunityId(TEST_HC_VAL);
+		assertionTypeTest.setHomeCommunity(homeCom);
+		msg.setAssertion(assertionTypeTest);
+		eventA.setMessage(msg);
+
+		CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+		verifycheckPolicy(result, direction);
+
+	}
+
+	private void verifycheckPolicy(CheckPolicyRequestType actualCheckPolicy, String direction) {
+
+		assertNotNull(actualCheckPolicy);
+
+		AssertionType assertTypeResult = actualCheckPolicy.getAssertion();
+
+		assertEquals("messageId should not change", MESSAGE_ID, assertTypeResult.getMessageId());
+
+		assertEquals("urn:oasis:names:tc:xacml:1.0:resource:resource-id",
+				actualCheckPolicy.getRequest().getResource().get(0).getAttribute().get(1).getAttributeId());
+		String expectResourceId = "amp;26.489.22";
+		assertEquals(expectResourceId, actualCheckPolicy.getRequest().getResource().get(0).getAttribute().get(0)
+				.getAttributeValue().get(0).getContent().get(0));
+
+		assertEquals(direction, actualCheckPolicy.getRequest().getAction().getAttribute().get(0).getAttributeValue()
+				.get(0).getContent().get(0));
+		SubjectType subjectType = actualCheckPolicy.getRequest().getSubject().get(0);
+		AttributeType attributeType = subjectType.getAttribute().get(1);
+		assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+		assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+		assertEquals(TEST_HC_VAL, attributeType.getAttributeValue().get(0).getContent().get(0));
+	}
+
+	@Test
+	public void testTransformAdhocQueryToCheckPolicy_InValidDirection() {
+		AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+		eventA.setDirection("in_outbound");
+		CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+		assertNull("We should expect null checkPolicyEngine ", result);
+		eventA.setDirection(null);
+		assertNull(instance.transformAdhocQueryToCheckPolicy(eventA));
+		assertNull(instance.transformAdhocQueryToCheckPolicy(null));
+	}
+
+	@Test
+	public void testTransformAdhocQueryToCheckPolicy_DefaultOutBound() {
+
```

Comment:
I don't understand the purpose of the "default" tests.  they don't test anything.